docs/error.md

```diff
@@ -0,0 +1,60 @@
+# Error
+
+An object representing an error that can be triggered / thrown.
+
+
+## **Overview**
+
+The errors are used to notify the user of a critical issue or log a recoverable one. An error object is consisted of three mandatory parameters: error severity, error category and an error code. The forth parameter is a mandatory object with more information about the issue.
```

Comment:
Errors are used to notify the user of a critical issue or to log information on a recoverable problem. An error object consists of three mandatory parameters: error severity, error category and an error code. The forth parameter is a mandatory object with additional information about the issue.